src/bin/cargo-flamegraph.rs

```diff
@@ -36,14 +36,16 @@ struct Opt {
     #[clap(long, group = "exec-args")]
     test: Option<String>,
 
-    /// Crate target to unit test, <unit-test> may be omitted if crate only has one target
-    /// (currently profiles the test harness and all tests in the binary; test selection
+    /// Crate target to unit test, <unit-test> may be omitted if the crate's library is targeted. To
```

Comment:
Can we preserve the property that we test whatever the target is if there is a single one? Seems silly to require that we specify the bin target if there is a single target.

Reply:
I know _this is silly_. The only way to keep the old behaviour (_I think_) requires major refactoring regarding the `find_single_target` function. If there are any future refactoring plans, I am up to the challenge.

Otherwise this is the best way to hot fix it.

Also to add to the explanation of #245, this is quite a major bug/missing feature, as it is really frequent to have a binary in `main.rs` that relies on the code of its library in `lib.rs`. There is currently no way of distinguishing between the binary and the library in this type of situation.

Reply:
I don't want this hotfix. Why is the required refactoring so "major"?